src/network/networkd-link.c

```diff
@@ -1683,10 +1707,13 @@ static int link_handle_bound_to_list(Link *link) {
 
         HASHMAP_FOREACH (l, link->bound_to_links, i)
                 if (link_has_carrier(l)) {
-                        required_up = true;
+                        required_up = link_should_activate(link);
```

Comment:
doesn't look right to me. This is the implementation of BindCarrier=. I am pretty sure that if BindCarrier= is used ActivationMode= should not have an effect (and we should probably log an error if it is set anyway). That's because BindCarrier= on its own already defines a rule when to up a link, and hence should be in control.